SU2_CFD/src/solvers/CSpeciesSolver.cpp

```diff
@@ -0,0 +1,724 @@
+
+    /*--- Detect a wrong solution file ---*/
+
+    if (counter != nPointDomain) {
+      SU2_MPI::Error(string("The solution file ") + restart_filename + string(" doesn't match with the mesh file!\n") +
+                         string("It could be empty lines at the end of the file."),
+                     CURRENT_FUNCTION);
+    }
+
+  }  // end SU2_OMP_MASTER, pre and postprocessing are thread-safe.
+  END_SU2_OMP_MASTER
+  SU2_OMP_BARRIER
+
+  /*--- MPI solution and compute the eddy viscosity ---*/
+
+  solver[MESH_0][SPECIES_SOL]->InitiateComms(geometry[MESH_0], config, SOLUTION);
+  solver[MESH_0][SPECIES_SOL]->CompleteComms(geometry[MESH_0], config, SOLUTION);
+
+  // Flow-Pre computes/sets mixture properties
+  solver[MESH_0][FLOW_SOL]->Preprocessing(geometry[MESH_0], solver[MESH_0], config, MESH_0, NO_RK_ITER,
+                                          RUNTIME_FLOW_SYS, false);
+  // Update eddy-visc which needs correct mixture density and mixture lam-visc. Note that after this, another Flow-Pre
+  // at the start of the Iteration sets the updated eddy-visc into the Flow-Solvers Primitives.
+  if (config->GetKind_Turb_Model() != TURB_MODEL::NONE)
+    solver[MESH_0][TURB_SOL]->Postprocessing(geometry[MESH_0], solver[MESH_0], config, MESH_0);
+  // For feature_multicomp this Scalar-Pre only computes the laminar contribution to mass diffusivity
+  solver[MESH_0][SPECIES_SOL]->Preprocessing(geometry[MESH_0], solver[MESH_0], config, MESH_0, NO_RK_ITER,
+                                             RUNTIME_FLOW_SYS, false);
+
+  /*--- Interpolate the solution down to the coarse multigrid levels ---*/
+
+  for (iMesh = 1; iMesh <= config->GetnMGLevels(); iMesh++) {
+    SU2_OMP_FOR_STAT(omp_chunk_size)
+    for (iPoint = 0; iPoint < geometry[iMesh]->GetnPoint(); iPoint++) {
+      Area_Parent = geometry[iMesh]->nodes->GetVolume(iPoint);
+      su2double Solution_Coarse[MAXNVAR] = {0.0};
+      for (iChildren = 0; iChildren < geometry[iMesh]->nodes->GetnChildren_CV(iPoint); iChildren++) {
+        Point_Fine = geometry[iMesh]->nodes->GetChildren_CV(iPoint, iChildren);
+        Area_Children = geometry[iMesh - 1]->nodes->GetVolume(Point_Fine);
+        Solution_Fine = solver[iMesh - 1][SPECIES_SOL]->GetNodes()->GetSolution(Point_Fine);
+        for (iVar = 0; iVar < nVar; iVar++) {
+          Solution_Coarse[iVar] += Solution_Fine[iVar] * Area_Children / Area_Parent;
+        }
+      }
+      solver[iMesh][SPECIES_SOL]->GetNodes()->SetSolution(iPoint, Solution_Coarse);
+    }
+    END_SU2_OMP_FOR
+
+    solver[iMesh][SPECIES_SOL]->InitiateComms(geometry[iMesh], config, SOLUTION);
+    solver[iMesh][SPECIES_SOL]->CompleteComms(geometry[iMesh], config, SOLUTION);
+
+    /*--- These solver Pre- Postprocessing calls should be consistent with the MESH_0 part above ---*/
+    solver[iMesh][FLOW_SOL]->Preprocessing(geometry[iMesh], solver[iMesh], config, iMesh, NO_RK_ITER, RUNTIME_FLOW_SYS,
+                                           false);
```

Comment:
I see, the Flow-solver already delegates that to the Turb-routine if Turb is present. The thing is the restart logic rn is not even necessary ... it will be for multicomponent flow from #1332 and as it was quite some debugging work I would like to get in a test before changing that (never trust yourself 🤨 )

Reply:
done in 156d0c5